docs/getting-started/handling-errors.md

```diff
@@ -0,0 +1,326 @@
+
+# Getting started: handling errors
+
+We've covered creating and throwing good errors, now let's talk about handling errors. The way you handle errors, particularly in Express, is important. You can end up writing a lot of code in different parts of your app if you're not careful. More code isn't always bad, but duplicated error handling can lead to your errors not being surfaced to where they're needed.
+
+  * [Bubbling up in Express](#bubbling-up-in-express)
+  * [Don't fear the try/catch block](#dont-fear-the-trycatch-block)
+    * [Error checking](#error-checking)
+    * [Multiple try/catch blocks](#multiple-trycatch-blocks)
+  * [Handling recoverable errors](#handling-recoverable-errors)
+
+
+## Bubbling up in Express
+
+In Express, error handlers can be bound at the Application and Route level. This allows you to write lots of error handling code in one place, and you can rely on your Express error handler to do things such as [log the appropriate error detail](./logging-errors.md), forward them on to a service like Splunk, or render a custom error page.
+
+The following is not a great pattern, as Express has no chance to log the error to a central place, we're missing a lot of information that we should be capturing:
+
+```js
+app.get('/', (request, response) => {
+    try {
+        exampleFunctionWhichThrowsAnError();
+    } catch (error) {
+        response.status(500).send(error.message);
+    }
+});
+```
+
+The way you make sure errors are handled by the main Express error handler is by always passing errors on. The way you do this will depend on whether your route handler is `async` or not.
+
+For non-`async` route handlers, any errors thrown will immediately be handled by Express:
+
+```js
+// This is better because Express will get hold of the error, potentially
+// log it, and render a page with a `500` status code
+app.get('/', (request, response) => {
+    exampleFunctionWhichThrowsAnError();
+});
+```
+
+For `async` route handlers, it's very important that you either use a module like [async-express-errors](https://www.npmjs.com/package/express-async-errors) (to avoid app crashes) or wrap your handler in an outer try/catch block:
```

Comment:
I see where you're coming from but I think because we can't _mandate_ that teams user `express-async-errors` I'd like to keep examples assuming that they don't. I think it might be worth adding an entirely new documentation page outside of the "getting started" guide which covers `async` route handlers and then linking to it from here.

I think I can add in a few clarifying messages to explain to the reader that we're making the assumption that they're not monkey-patching Express, and also that if they _are_ then they can ignore the `next` pattern.

Reply:
I've updated the docs to better explain this 👍  thanks. See the most recent commit! Feel free to unresolve if you think we need more